### PR TITLE
Support reading IP state

### PR DESCRIPTION
**Pull request by tyll**

### CODE REVIEW

**Comment by tyll** [comment]
This is just reading the IP state from #19

File: libnmstate/netinfo.py [hunk 1]
```diff
@@ -91,6 +96,30 @@ def _devinfo_bond(dev, devinfo):
             }
     return {}
 
+def _devinfo_ip(dev, devinfo):
+    ip4info = {}
+    connection = dev.get_active_connection()
```
Comment (EdDev) [inline libnmstate/netinfo.py]: Please move all NM related details to nm.* , here we should only be concern with the higher format,
Something like this:
```
def _devinfo_ipv4(dev):
    act_con = nm.connection.get_device_active_connection(dev)
    return {'ipv4': nm.ipv4.get_state(act_con)}
```

File: libnmstate/netinfo.py [hunk 2]
```diff
@@ -91,6 +96,30 @@ def _devinfo_bond(dev, devinfo):
...
+    if connection:
```
Comment (EdDev) [inline libnmstate/netinfo.py]: We can reduce indentation and make it easier to reach by exiting early:

if not act_connection:
    return {'enabled': False}
...

File: libnmstate/netinfo.py [hunk 3]
```diff
@@ -91,6 +96,30 @@ def _devinfo_bond(dev, devinfo):
...
+        ip4config = connection.get_ip4_config()
+        if ip4config:
+            ip4info['enabled'] = True
+            addresslist = []
+
+            addresses = ip4config.get_addresses()
+            for address in addresses:
```
Comment (EdDev) [inline libnmstate/netinfo.py]: This can be converted into a list comprehension.

File: libnmstate/nmclient.py [hunk 4]
```diff
@@ -39,3 +39,16 @@ def client(refresh=False):
     if _nmclient is None or refresh:
         _nmclient = NM.Client.new(None)
     return _nmclient
+
+
+def connection_ensure_setting(connection, setting_type):
```
Comment (EdDev) [inline libnmstate/nmclient.py]: This is used in the next patch?

Comment (tyll) [reply libnmstate/nmclient.py]: Sorry, this was left from before and can go.

**Comment by tyll** [comment]
This was modified in #34

**Pull request closed without merging**